pypeit/core/findobj_skymask.py

```diff
@@ -1762,9 +1762,16 @@ def objs_in_slit(image, ivar, thismask, slit_left, slit_righ,
     gpm_smash = npix_smash > 0.3*nsmash
     flux_sum_smash = np.sum((image_rect*gpm_sigclip)[find_min_max_out[0]:find_min_max_out[1]], axis=0)
     flux_smash = flux_sum_smash*gpm_smash/(npix_smash + (npix_smash == 0.0))
+    # Mask around the peaks
+    peak_bpm = arc.mask_around_peaks(flux_smash, np.logical_not(gpm_smash))
+    # Check that the peak mask is not empty
+    if np.sum(np.logical_not(peak_bpm))/peak_bpm.size < 0.05:  # 95% masked!
+        msgs.warn('Peak masking was too aggressive. Attempting to sigma clip.')
+        smash_mask = np.logical_not(gpm_smash)
+    else:
+        smash_mask = np.logical_not(gpm_smash) | peak_bpm
     flux_smash_mean, flux_smash_med, flux_smash_std = astropy.stats.sigma_clipped_stats(
-        flux_smash, mask=np.logical_not(gpm_smash), sigma_lower=3.0, sigma_upper=3.0
-    )
+        flux_smash, mask=smash_mask, sigma_lower=3.0, sigma_upper=3.0)
```

Comment:
I haven't dug into the context of this change, but it feels important.  Can you give us more information about the problem that you've addressed with this code?  And, do you have any assessments of how this affects all the other instruments?

Reply:
To be honest, I developed this many months ago, and I can't remember the motivation, or why it wasn't working in the first place. The code is basically finding peaks and masking around them, provided that pixel values continuously decrease away from the peak pixel. There needs to be a continuous decrease of 5+ pixels in order to activate this feature. I suspect this is only important when object profiles are excessively broad (which is what I was dealing with at the time).

Given that I can't recall what the motivation was, I am happy to revert this.